docs/proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,319 @@
+      AvailableBytes *uint64 `json:"availablebytes,omitempty"`  
+    }  
+
+    // DiskResources contains data about filesystem disk resources.  
+    type DiskResources struct {  
+      // The time at which these stats were updated.  
+      Time metav1.Time `json:"time"`  
+      // The device that this filesystem is on  
+      Device string `json:"device"`  
+      // AvailableBytes represents the storage space available (bytes) for the filesystem.  
+      // +optional  
+      AvailableBytes *uint64 `json:"availableBytes,omitempty"`  
+      // CapacityBytes represents the total capacity (bytes) of the filesystems underlying storage.  
+      // +optional  
+      CapacityBytes *uint64 `json:"capacityBytes,omitempty"`  
+      // InodesFree represents the free inodes in the filesystem.  
+      // +optional  
+      InodesFree *uint64 `json:"inodesFree,omitempty"`  
+      // Inodes represents the total inodes in the filesystem.  
+      // +optional  
+      Inodes *uint64 `json:"inodes,omitempty"`  
+    }  
+
+    // PodUsage holds pod-level unprocessed sample stats.  
+    type PodUsage struct {  
+      // UID of the pod  
+      PodUID string `json:"uid"`  
+      // Stats pertaining to pod total usage of cpu  
+      // This may include additional overhead not included in container usage statistics.  
+      // +optional  
+      CPU *CpuUsage `json:"cpu,omitempty"`  
+      // Stats pertaining to pod total usage of system memory  
+      // This may include additional overhead not included in container usage statistics.  
+      // +optional  
+      Memory *MemoryUsage `json:"memory,omitempty"`  
+      // Stats of containers in the pod.  
+      Containers []ContainerUsage `json:"containers" patchStrategy:"merge" patchMergeKey:"uid"`  
+      // Stats pertaining to volume usage of filesystem resources.  
+      // +optional  
+      Volumes []VolumeUsage `json:"volume,omitempty" patchStrategy:"merge" patchMergeKey:"name"`  
+    }  
+
+    // ContainerUsage holds container-level usage stats.  
+    type ContainerUsage struct {  
+      // UID of the container  
+      ContainerUID string `json:"uid"`  
+      // Stats pertaining to container usage of cpu  
+      // +optional  
+      CPU *CpuUsage `json:"memory,omitempty"`  
+      // Stats pertaining to container usage of system memory  
+      // +optional  
+      Memory *MemoryUsage `json:"memory,omitempty"`  
+      // Stats pertaining to container rootfs usage of disk.  
+      // Rootfs.UsedBytes is the number of bytes used for the container write layer.  
+      // +optional  
+      Rootfs *DiskUsage `json:"rootfs,omitempty"`  
+      // Stats pertaining to container logs usage of Disk.  
+      // +optional  
+      Logs *DiskUsage `json:"logs,omitempty"`  
+    }  
+
+    // CpuUsage holds statistics about the amount of cpu time consumed  
+    type CpuUsage struct {  
+      // The time at which these stats were updated.  
+      Time metav1.Time `json:"time"`  
```

Comment:
this should probably be `Timestamp` (and elsewhere in this proposal) so it's not confused with CPU time (plus, we use `Timestamp` elsewhere IIRC)